src/backend/distributed/transaction/backend_data.c

```diff
@@ -826,6 +822,21 @@ AssignGlobalPID(void)
 
 	SpinLockAcquire(&MyBackendData->mutex);
 	MyBackendData->globalPID = globalPID;
+	MyBackendData->distributedCommandOriginator = distributedCommandOriginator;
+	SpinLockRelease(&MyBackendData->mutex);
+}
+
+
+/*
+ * OverrideBackendDataWorkerQuery should only be used for isolation testing.
+ * See how it is used in the relevant functions.
+ */
+void
+OverrideBackendDataWorkerQuery(bool distributedCommandOriginator)
```

Comment:
This is for overriding the `distributedCommandOriginator` which is the opposite of `WorkerQuery`.
Can you rename to `OverrideBackendDataDistributedCommandOriginator`?

Reply:
Ops, there is a confusion between WorkerQuery and distributedCommandOriginator. I'll make all distributedCommandOriginator.